src/proof/alethe/alethe_printer.cpp

```diff
@@ -0,0 +1,265 @@
+    {
+      Trace("alethe-printer") << "... found assumption in list "
+                              << ": " << args[2] << "/" << assumptions
+                              << "     " << it->second << std::endl;
+      return it->second;
+    }
+
+    Trace("alethe-printer") << "... printing failed! Encountered assumption "
+                               "that has not been printed! "
+                            << args[2] << "/" << assumptions << std::endl;
+    return "";
+  }
+
+  // Print children
+  std::vector<std::string> child_prefixes;
+  std::string prefix = current_prefix;
+  if (current_prefix != "" && current_prefix.back() != '.')
+  {
+    prefix.append(".");
+  }
+  for (const std::shared_ptr<ProofNode> child : pfn->getChildren())
+  {
+    child_prefixes.push_back(
+        printInternal(out, child, assumptions, steps, prefix, current_step_id));
+  }
+
+  // If the rule is a subproof a final subproof step needs to be printed
+  if (arule == AletheRule::ANCHOR_SUBPROOF || arule == AletheRule::ANCHOR_BIND)
+  {
+    Trace("alethe-printer") << "... print anchor node " << pfn->getResult()
+                            << " " << arule << " / " << args << std::endl;
+
+    out << "(step " << current_prefix << " " << args[2] << " :rule " << arule;
+
+    // Reset step id to the number before the subproof + 1
+    current_step_id = step_id + 1;
+
+    // Discharge assumptions in the case of subproof
+    // The assumptions of this level have been stored in current_assumptions
+    if (arule == AletheRule::ANCHOR_SUBPROOF)
+    {
+      out << " :discharge (";
+      for (unsigned long int j = 0; j < current_assumptions.size(); j++)
+      {
+        out << current_assumptions[j];
+        if (j != current_assumptions.size() - 1)
+        {
+          out << " ";
+        }
+      }
+      out << ")";
+    }
+    out << ")\n";
+    return current_prefix;
+  }
+
+  // If the current step is already printed return its id
+  auto it = steps.find(args[2]);
+
+  if (it != steps.end())
+  {
+    Trace("alethe-printer")
+        << "... step is already printed " << pfn->getResult() << " " << arule
+        << " / " << args << std::endl;
+    return it->second;
+  }
+
+  // Print current step
+  Trace("alethe-printer") << "... print node " << pfn->getResult() << " "
+                          << arule << " / " << args << std::endl;
+  std::string current_t =
+      current_prefix + "t" + std::to_string(current_step_id);
+  out << "(step " << current_t << " ";
+  out << args[2] << " :rule " << arule;
+  current_step_id = current_step_id + 1;
+  if (args.size() > 3)
+  {
+    out << " :args (";
+    for (unsigned long int i = 3, size = args.size(); i < size; i++)
+    {
+      if (arule == AletheRule::FORALL_INST)
+      {
+        out << "(:= " << args[i][0] << " " << args[i][1] << ")";
+      }
+      else
+      {
+        out << args[i];
+      }
+      if (i != args.size() - 1)
+      {
+        out << " ";
+      }
+    }
```

Comment:
```suggestion
    for (size_t i = 3, size = args.size(); i < size; i++)
    {
      if (arule == AletheRule::FORALL_INST)
      {
        out << "(:= " << args[i][0] << " " << args[i][1] << ")";
      }
      else
      {
        out << args[i];
      }
      if (i != size - 1)
      {
        out << " ";
      }
    }
```

But for the `AletheRule::FORALL_INST` case distinction there there is a better way to handle this, which will allow the printing to be uniform: in the post processor, when processing FORALL_INST, you can transform its list of arguments into a list of arguments where each element is of the form `(SEXPR ":=" args[i][0] args[i][1])`. This way you can simplify the printer.

Reply:
After discussion this with @HanielB it came out that there is some work elsewhere needed to print ":=" as := and not as |:=|.  Thus, we decided to keep this handling for now and change it to the more convenient way in a separate PR.